### PR TITLE
cloud: update azure sdk for br

### LINKED ISSUE #14851: update Azure SDK for br
## Development Task
Due to the tikv updating the Azure SDK, we can:
1. make the `AzureStorage` compatible with the new Azure SDK.
2. introduce the [`encryption-scope`](https://learn.microsoft.com/en-us/azure/storage/blobs/encryption-scope-manage?tabs=powershell#upload-a-blob-with-an-encryption-scope) and [`sas`](https://learn.microsoft.com/en-us/rest/api/storageservices/delegate-access-with-shared-access-signature) for br.

### PR DESCRIPTION
**Pull request by Leavrth**
### What is changed and how it works?

Issue Number: Close #14851 

What's Changed:
1. make the AzureStorage compatible with the new Azure SDK.
2. introduce the [encryption-scope](https://learn.microsoft.com/en-us/azure/storage/blobs/encryption-scope-manage?tabs=powershell#upload-a-blob-with-an-encryption-scope) and [sas](https://learn.microsoft.com/en-us/rest/api/storageservices/delegate-access-with-shared-access-signature) for br.

```commit-message
```

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

1. test using the `sas` to access blobs.
2. test using the `encryption-scope` to encrypt blobs in the server side.

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/cloud/azure/src/azblob.rs [hunk 1]
```diff
@@ -225,47 +236,35 @@ impl BlobConfig for Config {
 
 enum RequestError {
     InvalidInput(Box<dyn std::error::Error + Send + Sync>, String),
-    InternalError(String),
```
Comment (Leavrth) [inline components/cloud/azure/src/azblob.rs:228]: the new azure sdk supports the retry backoff, so no need to wrap the another retry anymore.

**Review by 3pointer — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**